samples/net/sockets/can/src/main.c

```diff
@@ -176,6 +180,10 @@ static int setup_socket(void)
 		return -ENOENT;
 	}
 
+#ifdef CONFIG_SAMPLE_SOCKET_CAN_LOOPBACK_MODE
+	can_set_mode(DEVICE_DT_GET(DT_CHOSEN(zephyr_canbus)), CAN_LOOPBACK_MODE);
```

Comment:
With loopback mode, we should be able to remove the `harness` from `samples/net/sockets/can/sample.yaml` and add a console fixture for validating the sample in CI.